### ISSUE TITLE
Add other option to 'When Done'

### ISSUE THREAD

**Issue by NotSamuel** [RandomEngy/VidCoder#801]
Hi, first of all  Great Program!

Second, would it be possible to add a custom option to the 'When Done' tab. ie When Done...'Run Program X'

As i would like to run another program upon encode completion. (In this case, a .bat file, to write the missing tags for MKV files)

Also, perhaps someone can help. I can't find in the documention, with 'Foreign Audio Search', does Vidcoder scan all files in the queue for subtitles containing foreign subs? Or does this feature only work for a single video file?

Thanks!

**Comment by RandomEngy** [comment]
Check the Picker window:

![image](https://user-images.githubusercontent.com/9450861/112381589-fb641780-8ca7-11eb-824a-f87738a4cef8.png)

Foreign Audio Search is an option that might be enabled for some queued items and disabled for others. It depends on if you had it enabled when you queued the job.

**Comment by NotSamuel** [comment]
Thankyou for the reply! Unfortunatley i can't seem to get the picker option working.

When 'post encoding' is selected, It does appear to 'run' my batch file, but the batch file, (which normaly processes each file in the folder, writing new tags for each one) does not register any files in the folder, despite the recently re-coded file definately being there.

I used the 'Browse' button to select the path of my batch file (which is located in the destination folder for recoded files).

"C:\Folder\WriteNewTags.bat"

I've no idea what arguments to put?

**Comment by quebecman** [comment]
This was already asked and I think it's under development.

-- In the PICKER to run a script after a JOB QUEUE (See ticket #769)

**Comment by RandomEngy** [comment]
It can't directly execute a batch file. You might have to try passing that batch file as an argument to cmd.

**Comment by NotSamuel** [comment]
I did try yesterday to pass %1 and %* to my batch file, without success. (i'm not a programmer, just a tinkerer).

It's not a big deal tbh. Once i've  re-coded a file, i like to re-write it's tags for when i use MediaInfo, as something (probably ffpmeg) doesn't write them properly for MKV.

I'll just have to do it manually. 

Thanks again for the prompt repsones.

**Comment by RandomEngy** [comment]
Try this:

![image](https://user-images.githubusercontent.com/9450861/112708588-0e294880-8e70-11eb-9ab7-e05dbddaa81e.png)

**Comment by NotSamuel** [comment]
Hello again!
After trying your suggestion on my system. I've found the following.

1) I needed to use 'cmd.exe' rather than 'cmd'. In the 'Send file to:' box
2) Batch files are the work of the devil :)
3) I needed to change a line in the batch file to reflect the current directory it was in. 

I am assuming that when i normally double-click the batch file to run it, it 'knows' where it is (and therefore 'sees' the files in that DIR), but when it is called from 'somewhere...ie...Vidcoder' it doesn't.

Therefore i included '%~dp0' (current directory) in the for loop which scans the directory for mkv files. Like so..

 for %%m in (_%~dp0_*.mkv) do ( 

Also, as my batch uses delayedexpansion (just to add a file counter!), file paths with unusual characters don't function as you might expect! This tripped me up for some time...

This now works perfectly. Thankyou!

**Comment by NotSamuel** [comment]
Ok, so i discovered that if i had more than 1 file in queue, the batch file would keep re-writing the tags of every file accumulating in the output folder. And while it only take 7 seconds tp write the tags for a reasoanbly large file (on my ssd) it could take longer on a spinning disk, probably uses 'some' cpu time, and is therefore not desirable. So..
I discarded the for loop, as Vidcoder executes the 'send file to' after every single file encode anyway.
And placed {file} at the end of the arguments.

 /c C:\folder1\test.bat {file}

And then used the variable '%1' in the batch file to pass the filename to MkvPropedit.

mkvpropedit  %1 --chapters ""   --tags all: --add-track-statistics-tags 

I set the batch file to run minimzed so it's not even seen.
Job done.

Once again, thanks for this marvelous program!
